syz-manager/http.go

```diff
@@ -325,22 +326,26 @@ func (mgr *Manager) httpCoverCover(w http.ResponseWriter, r *http.Request, funcF
 		return
 	}
 
+	mgr.mu.Lock()
+	rgCache := mgr.reportGenerator
+	mgr.mu.Unlock()
+
 	// Don't hold the mutex while creating report generator and generating the report,
 	// these operations take lots of time.
-	if !mgr.checkDone.Load() {
+	if !mgr.checkDone.Load() || rgCache == nil {
```

Comment:
I would assume if mgr.checkDone is set, then mgr.reportGenerator should be initialized already. Isn't it the case?

We have a mutex-protected lazy cache of a mutex-protected lazy cache. This looks over-complicated, hard to use, and brittle.

If we introduce ReportGeneratorWrapper, can we make it non-lazy?
Namely, ReportGeneratorWrapper is created on start when Manager object is created, but w/o modules. ReportGeneratorWrapper.Get is always safe to use and does not require any external synchronization, it may just return nil.
When we have modules we call ReportGeneratorWrapper.Init and pass modules. After that ReportGeneratorWrapper.Get will do lazy init and return non-nil ReportGenerator.

This will remove one level of mutex-protection and laziness, and should simplify uses.

Reply:
Good idea!
I did it using `ReportGenerator.SetModules()`.